http/http.go

```diff
@@ -18,9 +20,89 @@ import (
 
 var tracer = trace.GlobalTracer
 
-// shared code for POSTing to an endpoint, that consumes JSON, that is zlib-
+// HTTPClientTracer is a wrapper around a `net/http/httptrace` that handles
+// proper tracing and metric emission.
+type HTTPClientTracer struct {
```

Comment:
So one thing that I notice here is that none of the structs and methods you introduce here need to be exported; not sure if you want to action, either's fine by me, but I'd err towards not exporting them so we can adjust the interface more easily as needed.